osu.Game/Online/API/Requests/Responses/APICommentsController.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Newtonsoft.Json;
+using osu.Game.Users;
+using System.Collections.Generic;
+
+namespace osu.Game.Online.API.Requests.Responses
+{
+    public class APICommentsController
+    {
+        private List<Comment> comments;
+
+        [JsonProperty(@"comments")]
+        public List<Comment> Comments
+        {
+            get => comments;
+            set
+            {
+                comments = value;
+                comments.ForEach(child =>
+                {
+                    if (child.ParentId != null)
+                    {
+                        comments.ForEach(parent =>
+                        {
+                            if (parent.Id == child.ParentId)
+                            {
+                                parent.ChildComments.Add(child);
+                                child.ParentComment = parent;
+                            }
+                        });
+                    }
+                });
+            }
+        }
+
+        [JsonProperty(@"has_more")]
+        public bool HasMore { get; set; }
+
+        [JsonProperty(@"has_more_id")]
+        public long? HasMoreId { get; set; }
+
+        [JsonProperty(@"user_follow")]
+        public bool UserFollow { get; set; }
+
+        [JsonProperty(@"included_comments")]
+        public List<Comment> IncludedComments { get; set; }
+
+        private List<User> users;
+
+        [JsonProperty(@"users")]
+        public List<User> Users
+        {
+            get => users;
+            set
+            {
+                users = value;
+
+                value.ForEach(u =>
+                {
+                    Comments.ForEach(c =>
+                    {
+                        if (c.UserId == u.Id)
+                            c.User = u;
+
+                        if (c.EditedById == u.Id)
+                            c.EditedUser = u;
+                    });
+                });
```

Comment:
Same comment about performance/time complexity as above applies here. This is *sort of* better in that it will regress to quadratic only in threads where there are as many users as comments, but still could be faster.